roles/tripleo_cluster_osds/tasks/add.yml

```diff
@@ -7,13 +7,9 @@
   when: "{{ devices | length == 0 }}"
 
 - name: Apply OSDs using the device list
-  shell: "{{ ceph_cli }} orch daemon add osd {{ item }}:{{ devices|join(',') }} --force"
-  register: orch_dev_zap
+  shell: "{{ ceph_cli }} orch daemon add osd {{ item }}:{{ devices|join(',') }}"
```

Comment:
did you consider using `ceph orch apply osd ... ?

Reply:
you mean something like: `ceph orch apply osd -i spec.yml` ?
This is covered here [1] and the idea is to improve the osd section including more filters and options (as described here [2]).
The spec template is still WIP but the long term strategy is to:

1. bootstrap a minimal cluster
2. render the spec (combining hostmap and service definition)
3. apply the spec to add the remaining resources and services to the cluster

[1] https://github.com/fmount/tripleo-ceph/blob/ceph_spec_apply/roles/tripleo_cluster_spec_render/templates/spec.yaml.j2#L16-L21

[2] https://docs.ceph.com/docs/master/cephadm/drivegroups

Reply:
sounds good to me!